react-basic-practice/src/pages/dashboard.tsx

```diff
@@ -49,21 +61,50 @@ const Dashboard: React.FC = () => {
     },
   ];
 
+  const handleEdit = (id: number) => {
+    setModalHeaderText("Edit Customer");
+    setSelectedCustomerId(id);
+    onOpen();
+  };
+
   return (
     <Box>
       <Box
-      backgroundColor= "linkWater"
+        backgroundColor="linkWater"
         display="flex"
         justifyContent="space-between"
         width="100%"
         padding="16px 20px"
       >
         <Search />
-        <Button label="+ Add Customer" variant="shadow" onClick={onOpen} />
+        <Button
+          label="+ Add Customer"
+          variant="shadow"
+          onClick={() => {
+            setModalHeaderText("Add Customer");
+            setSelectedCustomerId(null);
+            onOpen();
+          }}
+        />
       </Box>
       {/* Table demo */}
-      <Table columns={TableColumn} data={data} />
-      <Modal isOpen={isOpen} onClose={onClose} />
+      <Table
+        columns={TableColumn.map((column) => ({
+          ...column,
+          value: (data: ICustomer) =>
```

Comment:
why do we define it here? I see you already defined it on TableColumn?

Reply:
i done remove duplicate code